bitmovin-analytics-datasource/src/components/QueryEditor.tsx

```diff
@@ -102,6 +102,11 @@ export function QueryEditor({ query, onChange, onRunQuery, datasource }: Props)
     onRunQuery();
   };
 
+  const onAliasByChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange({ ...query, aliasBy: event.target.value });
+    onRunQuery();
```

Comment:
Is it necessary to run the query again just for changing the alias? It is not part of the query right?

Reply:
I just tested it. Yes it is not part of the query put if we don't call the onRunQuery then the query method does not get called which also means that the alias does not get applied. So we need to call it here, even though we wouldn't need to query our API. Maybe we could check in the query method what attribute changed and not query our API when only the alias by changes, but honestly I don't think this is worth the effort 🤔